integration/Jenkinsfiles/sonar.groovy

```diff
@@ -25,37 +25,58 @@ properties([
+    ]],
     pipelineTriggers([[$class: 'GitHubPushTrigger']])
   ])
 
 node('SLAVE') {
+    def BRANCH = params.BRANCH
+    if (env.BRANCH_NAME) {
+        println "Using BRANCH_NAME parameter: $env.BRANCH_NAME"
+        BRANCH = env.BRANCH_NAME
+    }
+    def PARENT_BRANCH = params.PARENT_BRANCH
+    if (env.CHANGE_TARGET) {
+        println "Using CHANGE_TARGET parameter: $env.CHANGE_TARGET"
+        PARENT_BRANCH = env.CHANGE_TARGET
+    }
+    println "Testing branch '$BRANCH' with parent branch '$PARENT_BRANCH'"
+    currentBuild.setDescription("$BRANCH -> $PARENT_BRANCH")
     tool type: 'ant', name: 'ant-1.9'
     tool type: 'hudson.model.JDK', name: 'java-8-oracle'
     tool type: 'hudson.tasks.Maven$MavenInstallation', name: 'maven-3'
     timeout(time: 5, unit: 'HOURS') {
         timestamps {
             try {
                 stage('clone') {
+                    // Check remote branch existence and effective checkouted branch
+                    sh 'git ls-remote --exit-code git://github.com/nuxeo/nuxeo.git $BRANCH'
                     git branch: '$BRANCH', url: 'git://github.com/nuxeo/nuxeo.git'
                     sh """#!/bin/bash -xe
-                          ./clone.py $BRANCH -f $PARENT_BRANCH
+                        ./clone.py $BRANCH -f $PARENT_BRANCH
+                        source scripts/gitfunctions.sh
+                        shr -a git show -s --pretty=format:'%h%d'
+                        test `git rev-parse HEAD` = `git rev-parse $BRANCH`
                     """
                     sha = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
                 }
                 stage('analysis') {
                     withBuildStatus("mvn verify sonar", "https://github.com/nuxeo/nuxeo", sha, BUILD_URL) {
                         try {
-                            withEnv(['MAVEN_OPTS=-Xmx6g -server']) {
+                            withEnv(['MAVEN_OPTS=-Xms4g -Xmx8g -XX:-UseGCOverheadLimit -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=$WORKSPACE/target/gcOoO.hprof']) {
                                 withCredentials([usernamePassword(credentialsId: 'c4ced779-af65-4bce-9551-4e6c0e0dcfe5', passwordVariable: 'SONARCLOUD_PWD', usernameVariable: '')]) {
                                     if (params.BRANCH != params.PARENT_BRANCH) {
                                         TARGET_OPTION="-Dsonar.branch.target=${params.PARENT_BRANCH}"
                                     } else {
                                         TARGET_OPTION=""
                                     }
                                     sh """#!/bin/bash -ex
-                                        mvn clean verify sonar:sonar -Dsonar.login=$SONARCLOUD_PWD -Paddons,distrib,qa,sonar -Dsonar.branch.name=$BRANCH $TARGET_OPTION \
+                                        mvn -B -V clean verify sonar:sonar -Dsonar.login=$SONARCLOUD_PWD -Paddons,distrib,qa,sonar -Dsonar.branch.name=$BRANCH $TARGET_OPTION \
```

Comment:
`-V` should be removed right ?
Do we need the `-B` option ?

Reply:
no, yes
`-V` is to always "Display version information WITHOUT stopping build", that is useful
`-B` is "Run in non-interactive (batch) mode (disables output color)", that is preferred
Here is the produced output:
```
+ mvn -B -V (...)
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T16:41:47+00:00)
Maven home: /opt/build/tools/maven3
Java version: 1.8.0_171, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-104-generic", arch: "amd64", family: "unix"
```